doc/diagnostics/AddingTracepoints.md

```diff
@@ -0,0 +1,417 @@
+<!--
+Copyright (c) 2020, 2020 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Guidelines for adding `-Xtrace` tracepoints to OpenJ9
+
+## Table of Contents:
+- Rules for adding tracepoint definitions to .tdf files
+- Adding tracepoints to a module that already has some
+- Tracepoint Format Specifiers
+- J9TraceFormat.dat
+- Adding the first tracepoint to a dynamic module
+- Adding the first tracepoint to a static module
+- Advanced use #1 - testing at runtime whether your tracepoint is enabled
+- Advanced use #2 - adding assert type tracepoints
+- Parameters to Assertions/Tracepoints and Side-effects (A warning)
+ 
+
+## Rules for adding tracepoint definitions to .tdf files
+
+1. Always **add new tracepoints at the end of the .tdf file**.
+1. If a tracepoint's signature changes, **obsolete it and create a new one**. By signature we mean the types, order and total number of format specifiers in the Template. Changes to the Tracepoint Type, Overhead, Level and `NoEnv` parameters, and cosmetic changes to the text in the Template can be made without adding a new tracepoint.
```

Comment:
Should this be point '2.'? Viewing this document on the web it seems to be corrected. Alternatively, all numbered items should begin with '1.' (e.g. lines 54-56 below) for consistency.

Reply:
It doesn't matter to markdown, but changed to use `2.` here to avoid confusion.